### ISSUE TITLE
Bookmarks lost with the ignorenonframetext option

### ISSUE THREAD

**Issue by josephwright** [josephwright/beamer#90]
Hi all. This is a followup to issue #55 

The `ignorenonframetext` option also removes the bookmarks from the pdflatex output file. This is a change from old behavior and anyway this goes beyond the stated purpose.

Even adding `hyperref={bookmarks=true}` doesn't overcome this behavior.

----------------------------------------
- Bitbucket: https://bitbucket.org/rivanvx/beamer/issue/90
- Originally reported by: Anonymous
- Originally created at: 2010-10-26

**Comment by josephwright** [comment]
_From Vedran Miletić on 2010-10-26_

I again can't reproduce. Please provide a minimal example.

**Comment by josephwright** [comment]
_From Anonymous on 2010-11-01_

All right, here you are
````
\documentclass[ignorenonframetext,xcolor={table,svgnames}]{beamer} %% slides
%%%% Bookmarks appear if one uncomments the following line instead
%%\documentclass[xcolor={table,svgnames}]{beamer} %% slides
%%\documentclass[handout,xcolor={table,svgnames}]{beamer} %% handouts

\usepackage{times}

\mode<article> % only for the article version
{
  \usepackage{fullpage}
  \usepackage{pgf}
  \usepackage[%
  colorlinks        =false,%    
  urlcolor          =green,%
  plainpages        =false,%
  % pdfpagemode       =None,%
  pdfpagelabels     =true,%
  % pdfstartview      =FitH,%     % Page fits horizontally.
  % pdfpagelayout     =SinglePage,% One page at a time
  bookmarks         =true,%
  bookmarksnumbered =true,%
  bookmarksopen     =true,%
  % bookmarksopenlevel=1,%
  % pdfborder         = {0 0 0}%    % No border
  naturalnames%                   % for algorithm2e
  ]{hyperref}
}

\mode<presentation>
{ 
  %%\usetheme{Montpellier} % handout
  \usetheme{Frankfurt} % slides

  \useinnertheme{rounded} % circles
 
  \usecolortheme{rose}
  \usecolortheme{sidebartab}

  \setbeamerfont{title}{shape=\itshape,family=\rmfamily}
  \setbeamercovered{dynamic}
}

\mode<handout>{
  \usepackage{pgfpages}
  \pgfpagesuselayout{2 on 1}[a4paper,border shrink=2mm,landscape]

  \useoutertheme{infolines}

  \setbeamercolor{background canvas}{bg=black!2} 
}

\institute[BUG]{%
  BUG\\
  \url{anonymous@buggy.com}}
\date{\color{red}BUG\normalcolor}

%%%%%%%%%%%%%%%%%%%%%%% Preamble ends here %%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{document}

\begin{frame}
  \titlepage
\end{frame}

\section*{Lecture contents}
\begin{frame}
  \tableofcontents[pausesections]
\end{frame}

\section{Brief overview}
\subsection{Goals of this session}
\begin{frame}{Our goals}
  \begin{itemize}[<+->]
  \item Bug 1
  \item Bug 2
  \end{itemize}
\end{frame}

\subsection{Solutions}
\begin{frame}{Our solutions}
  \begin{itemize}[<+->]
  \item Patch 1
  \item Patch 2
  \end{itemize}
\end{frame}

\section{Beamer everywhere}
\subsection{Good doc}
\begin{frame}{Good doc}
  \begin{itemize}
  \item Manual
  \item Tutorials
  \item Examples
  \end{itemize}
\end{frame}

\subsection{Only problem}
\begin{frame}{New themes are a pain ...}
  \begin{itemize}[<+->]
  \item Configuration spread everywhere
  \item No theme template
  \end{itemize}
\end{frame}

\end{document}
````

**Comment by josephwright** [comment]
_From Anonymous on 2011-03-01_

Hi,

Could u reproduce the abovementioned behavior or not?

**Comment by josephwright** [comment]
_From Anonymous on 2011-05-16_

Hi again, a new update.

Seems to be a regression bug introduced some time along the way. It seems to be a bad interaction between the starred version of `\section` and the `ignorenonframetext` option of `beamer`. Could you fix this now?

Thanks.

**Comment by josephwright** [comment]
_From Joseph Wright on 2011-09-24_

Okay, I've narrowed this down. Using `\section*` is redefining `\Hy@writebookmark` to gobble the input. For some reason, this is happening globally. Hopefully I'm close to sorting this!

**Comment by josephwright** [comment]
Fixed by d81eaceec271327f0f9dc9a6cdfa79dcb2eaa69d

**Comment by josephwright** [comment]
_From Anonymous on 2011-09-26_

Hi,

Thanks for the fix. It worked.

One point though: there is a glitch in the way the zip (and possibly gz and bz2 as well) downloadable source archive is built. The RCS expansion didn't take place so a drop-in replacement of `beamerbasecompatibility.sty` in my texmf tree results in a failure. I had to manually expand `$Header: $` (see line 11 of said `.sty` file).

I don't know if `hg archive` is enough with the `hg keyword` extension enabled.
